core/test/Test/Pos/Core/EnumEmpTest.hs

```diff
@@ -0,0 +1,111 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+module Test.Pos.Core.EnumEmpTest
+       ( tests
+       ) where
+
+-- The types `EpochOrSlot`, `LocalSlotIndex` and `SlotId` have a
+-- `HasProtocoConstants` constraint as well as `Enum` instances which depend
+-- on the `HasProtocoConstants` constraint. Since we want to remove
+-- `HasProtocoConstants` the first step is to provide functions that will act
+-- like the `Enum` methods, but take an extra parameter for
+-- the part they require from `ProtocolConstants`.
+
+-- These tests to show the equivalence between the 'Enum/HasProtocolConstant'
+-- instances for these types and the new functions that replace them. When the
+-- 'Enum/HasProtocolConstant' are removed, these tests will also be removed.
+
+import           Control.DeepSeq (force)
+import           Control.Exception (ErrorCall (..))
+import qualified Control.Exception as Exception
+import           Control.Monad.IO.Class (liftIO)
+import           Universum
+
+import           Hedgehog (Property, PropertyT, discover, (===))
+import qualified Hedgehog as H
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+import           Pos.Core (pcEpochSlots, withProtocolConstants)
+import           Pos.Core.Slotting (epochOrSlotFromEnum, epochOrSlotToEnum,
+                     localSlotIndexFromEnum, localSlotIndexToEnum,
+                     slotIdFromEnum, slotIdToEnum)
+
+import           Test.Pos.Core.Gen
+
+
+
+prop_enum_eqivanent_epochOrSlotFromEnum :: Property
+prop_enum_eqivanent_epochOrSlotFromEnum =
+  H.withTests 5000 . H.property $ do
```

Comment:
Shouldn't we indent by 4 spaces here or am I mistaken? (This applies to all the props)

Reply:
Yes we should. Thanks.